pkg/network/tracer/tracer_linux_test.go

```diff
@@ -1708,13 +1707,18 @@ func TestTCPDirectionWithPreexistingConnection(t *testing.T) {
 	require.NoError(t, err)
 	wg.Wait()
 
+	var origConn []network.ConnectionStats
 	// the original connection should still be incoming for the server
-	conns := getConnections(t, tr)
-	origConn := searchConnections(conns, func(cs network.ConnectionStats) bool {
-		return fmt.Sprintf("%s:%d", cs.Source, cs.SPort) == server.address &&
-			fmt.Sprintf("%s:%d", cs.Dest, cs.DPort) == c.LocalAddr().String()
-	})
-	require.Len(t, origConn, 1)
+	require.Eventually(t, func() bool {
+		conns := getConnections(t, tr)
+		origConn = searchConnections(conns, func(cs network.ConnectionStats) bool {
```

Comment:
Does it make sense to have searchConnections use require.Eventually itself? this is the second such fix I've seen that we've fixed by wrapping searchConnections in a require.Eventually